### PR TITLE
Added React JS UI for button deleting task

### LINKED ISSUE #253: [Task]: Implement delete task UI on the /users/{user_id}/tasks React page
What is it for?
This task will create/implement a feature in which a user can select a  single task and click on the delete button to delete it.

Note: **DISCLAIMER**: I have not been able to get the react app to run.  I have tried npm start and npm start run. I even researched the problem on  stack overflow, which involved me checking the package.json file for both the app and the REACT UI, but  I haven't been able to find a concrete solution.  I am not sure if it is my machine or if it is a potential BUG.  Just a heads up.

**To test the PR:**
To test the button:
1.) Type npm start (either inside VSC or nodejs)
2.) Window will open in your default browser (Mines was set to Firefox)
3.) Navigate to users/{user_id}/tasks
4.) Add/Find an existing task
5.) Press the "Delete Task" button for that task.
6.) The task should be deleted at this point--to check, you can navigate to the tasks page and see if it has been deleted.

### PR DESCRIPTION
**Pull request by chonger878**
Closes  #253 

This PR:
-Has a React JS UI for a delete button
-This "delete button" would delete a task at the user's request

To test the button:
1.) Type npm start (either inside VSC or nodejs)
2.) Window will open in your default browser (Mines was set to Firefox)
3.) Navigate to users/{user_id}/tasks
4.) Add/Find an existing task
5.) Press the "Delete Task" button
6.) The task should be deleted at this point--to check, you can navigate to the tasks page and see if it has been deleted.

Note: DISCLAIMER: I have not been able to get the react app to run. I have tried npm start and npm start run. I even researched the problem on stack overflow, which involved me checking the package.json file for both the app and the REACT UI, but I haven't been able to find a concrete solution. I am not sure if it is my machine or if it is a potential BUG. Just a heads up.

### CODE REVIEW

**Comment by unicornsleuth** [comment]
Can you please provide a walkthrough of how to test this?

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js [hunk 1]
```diff
@@ -14,11 +15,20 @@ const UserTasks = () => {
...
     .catch((error) => console.error(error))
   }, [userId]);
 
+  function deleteTask(taskId){
+    const messages = this.state.messages.filter((_, index) => index !== i)
+    this.setState({ messages });
+
+  }
+
+  let deleteTask = deleteTask(taskId);
```
Comment (farhadbahrehmandhenry) [inline ui/react-ui/src/components/routes/userTasks/UserTasks.js]: I think you don't need this line

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js [hunk 2]
```diff
@@ -28,9 +38,12 @@ const UserTasks = () => {
         ? <BootstrapTable heatherItems={Object.keys(userTasks[2][0])} rows={userTasks[2]} /> 
         : <h3>User {userId} Tasks Not Found</h3>
       }
+      <button onClick={this.deleteTask.bind(this)}>Delete Task</button>
```
Comment (farhadbahrehmandhenry) [inline ui/react-ui/src/components/routes/userTasks/UserTasks.js]: I think this should be like this =>
<button onClick={() => this.deleteTask()}>Delete Task</button>

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js (hunk 1 above)
Comment (chonger878) [reply ui/react-ui/src/components/routes/userTasks/UserTasks.js]: Deleted the line.

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js (hunk 2 above)
Comment (chonger878) [reply ui/react-ui/src/components/routes/userTasks/UserTasks.js]: Fixed it. Thanks!

**Review by jaguiniga — APPROVED** [review]
Looks like the requests from other reviews was fixed and everything left looks ok.

**Review by kieranmiguel — APPROVED** [review]

**Pull request merged**